python/triton/language/core.py

```diff
@@ -2865,9 +2865,12 @@ def kernel(...):
     :param loop_unroll_factor: Tells the Triton IR level loop unroller how many
         times to unroll a for loop that this range is used with. Less than 2 for
         this value implies no unrolling.
+    :param fuse: automatically fuse the loop nest starting at this loop to
+        create a single fused loop. The compiler will try to pipeline the fused
+        loop which can avoid stage stalling.
```

Comment:
If we're going to put this in the frontend maybe we should fix the misnomer. Loop fusion is joining two independent loops, e.g.
```
for i in range(n):
    foo(i)
for i in range(n):
    bar(i)
// =>
for i in range(n):
    foo(i)
    bar(i)
```

This is really loop flattening. Perhaps `flatten_inner_loop` would be a more descriptive name?

Reply:
yeah it is a misnomer. I can fix the name everywhere